src/dtos/monitor-location-base.dto.ts

```diff
@@ -21,9 +28,9 @@ export class MonitorLocationBaseDTO {
   @IsOptional()
   @IsString()
   @MaxLength(6)
-  @MatchesRegEx('[A-z0-9 -*#]{1,6}', {
+  @MatchesRegEx('^[A-z0-9\\-\\*#]{1,6}$', {
     message: (args: ValidationArguments) => {
-      return `${args.property} [MONLOC-FATAL-A] The value : ${args.value} for ${args.property} must be match the RegEx: [A-z0-9 -*#]{1,6}`;
+      return `The value of [${args.value}] for [${args.property}] must be match the RegEx: [A-z0-9-*#]{1,6} for [${KEY}].`;
```

Comment:
I think if we are mentioning a regular expression in the error message it should be spelled out, instead of "RegEx".  Also, the ^ and $ are part of the regular expression, marking the beginning and ending of the string being matched, so they should be in the error message if it reports the regular expression.  I think the error message should actually be something more user friendly like "Value must be 1 to 6 characters and only consist of upper and lower case letters, numbers, and the special characters - (dash), * (asterisk), and # (pound).

Reply:
updated